cmd/fleetctl/user.go

```diff
@@ -172,6 +175,126 @@ func createUserCommand() *cli.Command {
 	}
 }
 
+func createBulkUsersCommand() *cli.Command {
+	return &cli.Command{
+		Name:  "import",
+		Usage: "Create bulk users",
+		UsageText: `This command will create a set of users in Fleet by importing a CSV file. Expected columns are: Name,Email,Password,SSO,API Only,Global Role,Teams. Created Users by default get Observer Role.
+		Password could be left blank, Users should receive an invitation link.`,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     csvFlagName,
+				Usage:    "csv file with all the users (required)",
+				Required: true,
+			},
+			configFlag(),
+			contextFlag(),
+			yamlFlag(),
+			debugFlag(),
+		},
+		Action: func(c *cli.Context) error {
+			client, err := clientFromCLI(c)
+			if err != nil {
+				return err
+			}
+
+			csvFilePath := c.String(csvFlagName)
+
+			csvFile, err := os.Open(csvFilePath)
+			if err != nil {
+				return err
+			}
+			defer csvFile.Close()
+			csvLines, err := csv.NewReader(csvFile).ReadAll()
+
+			if err != nil {
+				return err
+			}
+			for _, record := range csvLines[1:] {
+				name := record[0]
+				email := record[1]
+				password := record[2]
+				sso := c.Bool(record[3])
+				apiOnly := c.Bool(record[4])
+				globalRoleString := c.String(record[5])
+				teamStrings := c.StringSlice(record[6])
+
+				var globalRole *string
+				var teams []fleet.UserTeam
+				if globalRoleString != "" && len(teamStrings) > 0 {
+					return errors.New("Users may not have global_role and teams.")
+				} else if globalRoleString == "" && len(teamStrings) == 0 {
+					globalRole = ptr.String(fleet.RoleObserver)
+				} else if globalRoleString != "" {
+					if !fleet.ValidGlobalRole(globalRoleString) {
+						return fmt.Errorf("'%s' is not a valid team role", globalRoleString)
+					}
+					globalRole = ptr.String(globalRoleString)
+				} else {
+					for _, t := range teamStrings {
+						parts := strings.Split(t, ":")
+						if len(parts) != 2 {
+							return fmt.Errorf("Unable to parse '%s' as team_id:role", t)
+						}
+						teamID, err := strconv.Atoi(parts[0])
+						if err != nil {
+							return fmt.Errorf("Unable to parse team_id: %w", err)
+						}
+						if !fleet.ValidTeamRole(parts[1]) {
+							return fmt.Errorf("'%s' is not a valid team role", parts[1])
+						}
+
+						teams = append(teams, fleet.UserTeam{Team: fleet.Team{ID: uint(teamID)}, Role: parts[1]})
+					}
+				}
+
+				if sso && len(password) > 0 {
```

Comment:
From 251 to 291 there's code duplication with `createUserCommand`. You could move this functionality to a "createUser" function and use them in both places (in `createUserCommand` and here `createBulkUsersCommand`), if they differ in some behavior you can add parameters to it.